code/modules/paperwork/desk_bell.dm

```diff
@@ -0,0 +1,108 @@
+			context[SCREENTIP_CONTEXT_LMB] = "Fix"
+
+	else
+		var/click_context = "Ring"
+		if(prob(1))
+			click_context = "Annoy"
+		context[SCREENTIP_CONTEXT_LMB] = click_context
+	return CONTEXTUAL_SCREENTIP_SET
+
+/obj/structure/desk_bell/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(!COOLDOWN_FINISHED(src, ring_cooldown) && ring_cooldown_length)
+		return TRUE
+	if(!ring_bell(user))
+		to_chat(user, span_notice("[src] is silent. Some idiot broke it."))
+	if(ring_cooldown_length)
+		COOLDOWN_START(src, ring_cooldown, ring_cooldown_length)
+	return TRUE
+
+/obj/structure/desk_bell/attackby(obj/item/weapon, mob/living/user, params)
+	. = ..()
+	times_rang += weapon.force
+	ring_bell(user)
+
+// Fix the clapper
+/obj/structure/desk_bell/screwdriver_act(mob/living/user, obj/item/tool)
+	if(broken_ringer)
+		user.visible_message(span_notice("[user] starts repairing [src]..."), span_notice("You start repairing [src]..."))
+		tool.play_tool_sound(src)
+		if(tool.use_tool(src, user, 50))
+			user.visible_message(span_notice("[user] repairs [src]!"), span_notice("You repair the clapper inside [src]."))
+			broken_ringer = FALSE
+			times_rang = 0
+			return TOOL_ACT_TOOLTYPE_SUCCESS
+		return FALSE
+	..()
+
+// Deconstruct
+/obj/structure/desk_bell/wrench_act_secondary(mob/living/user, obj/item/tool)
+	user.visible_message(span_notice("[user] starts taking [src] apart..."), span_notice("You start taking [src] apart..."))
+	tool.play_tool_sound(src)
+	if(tool.use_tool(src, user, 50))
+		user.visible_message(span_notice("[user] disassembles [src]!"), span_notice("You break down [src] into scrap metal."))
+		playsound(user, 'sound/items/deconstruct.ogg', 50, TRUE)
+		if(!broken_ringer) // Drop 2 if it's not broken.
+			new/obj/item/stack/sheet/iron(drop_location())
+		new/obj/item/stack/sheet/iron(drop_location())
+		qdel(src)
+		return TOOL_ACT_TOOLTYPE_SUCCESS
+	..()
+
+/// Check if the clapper breaks, and if it does, break it
+/obj/structure/desk_bell/proc/check_clapper(mob/living/user)
+	if((times_rang >= 10000) || prob(times_rang/100))
+		to_chat(user, span_notice("You hear [src]'s clapper fall off of its hinge. Nice job, you broke it."))
+		broken_ringer = TRUE
+
+/// Ring the bell
+/obj/structure/desk_bell/proc/ring_bell(mob/living/user)
+	if(!broken_ringer)
+		check_clapper(user)
+		// The lack of varying is intentional. The only variance occurs on the strike the bell breaks.
+		playsound(src, ring_sound, 70, vary = broken_ringer, extrarange = SHORT_RANGE_SOUND_EXTRARANGE)
+		flick("desk_bell_ring", src)
+		times_rang++
+		return TRUE
+	return FALSE
+
+// A warning to all who enter; the ringing sound STACKS. It won't be deafening because it only goes every decisecond,
+// but I did feel like my ears were going to start bleeding when I tested it with my autoclicker.
+/obj/structure/desk_bell/speed_demon
+	desc = "The cornerstone of any customer service job. This one's been modified for hyper-performance."
+	ring_cooldown_length = 0
```

Comment:
won't spam-clicking this break the bell pretty fast?

Reply:
If you think that'll be an issue then I'll do something to fix that when I wake up